src/lib/condition.js

```diff
@@ -234,11 +281,13 @@ class Condition {
    * @return {String} Condition as a URI
    */
   serializeUri () {
-    return 'cc' +
-      ':' + this.getTypeId().toString(16) +
-      ':' + this.getBitmask().toString(16) +
-      ':' + base64url.encode(this.getHash()) +
-      ':' + this.getMaxFulfillmentLength()
+    const ConditionClass = TypeRegistry.findByTypeId(this.type).Class
+    const includeSubtypes = ConditionClass.TYPE_CATEGORY === 'compound'
+    return 'ni:sha-256;' +
+      base64url.encode(this.getHash()) +
+      '?fpt=' + this.getTypeName() +
+      '&cost=' + this.getCost() +
+      (includeSubtypes ? '&subtypes=' + Array.from(this.getSubtypes()).sort().join(',') : '')
```

Comment:
That wouldn't work, because crypto conditions support looking up a condition by fulfillment. If it was stored with a different order, you might not find it or you'd need a more complicated indexing method than just indexing by condition. Fortunately, lexical sorting is trivial to implement in every language.